app/extensions/acm/__init__.py

```diff
@@ -35,12 +37,79 @@ class ACMManager(RestManager):
         'version': {
             'dict': '//version/',
         },
+        'passthrough': {
+            'data': '',
+        },
+        'job': {
+            'detect_request': '//engine/detect/%s',
+            'identification_request': '//engine/identify/%s',
+            'response': '//engine/job/metadata/?jobid=%s'
+        }
     }
     # fmt: on
 
     def __init__(self, pre_initialize=False, *args, **kwargs):
         super(ACMManager, self).__init__(pre_initialize, *args, **kwargs)
 
+    # The acm API returns a status and a response, this processes it to raise an exception on any
+    # error and provide validated parsed output for further processing
+    def request_passthrough_parsed(
+        self, tag, method, passthrough_kwargs, args=None, target='default'
+    ):
+        response = self.request_passthrough(tag, method, passthrough_kwargs, args, target)
+
+        # Sage sent invalid response
+        try:
+            response_json = response.json()
+        except Exception:
+            message = (f'{tag} {method} failed to parse json response from Sage',)
+            raise HoustonException(
+                status_code=400,
+                message=message,
+                log_message=f'{message} Sage Status:{response.status_code} Sage Reason: {response.reason}',
+            )
+
+        # Sage sent invalid response
+        status_data = response_json.get('status', None)
+        if not status_data:
+            message = (f'{tag} {method} failed to parse json status data from Sage',)
+            raise HoustonException(
+                status_code=400,
+                message=message,
+                log_message=f'{message} Sage Status:{response.status_code} Sage Reason: {response.reason}',
+            )
+
+        # status is correctly formatted, see if it failed
+        response_data = response_json.get('response', None)
+        if (
+            not response.ok
+            or not status_data.get('success', False)
+            or response.status_code != 200
+            or response_data is None
+        ):
+            log_message = status_data.get('message', response.reason)
+            #  Don't report internal Sage Errors to the frontend
+            message = 'failed to start detection'
+
+            status_code = response.status_code
+            if status_code > 600:
+                status_code = 400  # flask doesnt like us to use "invalid" codes. :(
```

Comment:
Should we append this code to the message so we have an easy avenue to inspect it?

Reply:
probably a good idea, especially given our (old-world, granted) tendency to hit the 6xx ones once in a while.  or at least maybe in the `log_message` ?

Reply:
That is what is passed as the acm_status_code in the exception below which should have been passed to the abort, but may not have been

Reply:
yup i think change for both, in the log message, would be great.  i didnt put it in the edm one because the edm side is outrageously verbose.  but that doesnt mean it would be useless to also get echoed in houston -- so lets go for that.